dashboard/server/index.js

```diff
@@ -2,12 +2,35 @@
 const socket = require('socket.io');
 const server = require('./server');
 
-const api = require('./api/index');
+const handlers = require('./api/index');
+const utils = require('../lib/utils');
 
 // setup socket.io server
 const io = socket(server);
 
+// middleware to extract id from the newman run
+io.use(utils.socketAuth);
```

Comment:
I think we're moving a lot of stuff to `utils`. It's fine for now, but ideally in a project, `utils` file should not exist at all.
If it does exist, it should only contain pure functions which perform some trivial tasks that shouldn't be repeated. They should not be aware of the business logic.